armi/runLog.py

```diff
@@ -98,82 +108,79 @@ class Log:
     Abstract class that ARMI code calls to be rendered to some kind of log.
     """
 
-    def __init__(self, verbosity=50):  # pylint: disable=unused-argument
+    def __init__(self, mpiRank=0, mpiSize=1):
```

Comment:
What real function does having our own Log class really serve at this point? I know we are currently using it to handle concatenation of different rank's log streams, but this could really be handled just about anywhere, and probably just be a straightforward function. We may wish to keep an adapter class around to maintain the API that folks are used to (`warn`, `info`, `extra`, etc.), but I even view that as optional, since currently no one (I don't think, anyway) interacts directly with the `Log` class; they're calling the module-level functions instead.

Reply:
I was actually maintaining the class partially _because_ I couldn't guarantee no one else would be subclassing it.

Since we have two different subclasses of it in `mockRunLogs.py` it seemed like the safe assumption would be to preserve the API as it is now.  BUT if the API doesn't involve the class, that's definitely something.

I would be happy to simplify the code, but let me look at it. Right now the class serves two purposes:

1. Easy encapsulation of log concatenation.
2. It handles the piping of all `stderr` messages into files.

Let me look at it.

Reply:
What API we land on is kind of wide open. The only thing that I think we _have_ to support (@drewejohnson let me know if you have anything that is different) are direct calls to `runLog.whichever()`. Beyond that, I'm considering this a clean sheet. Direct access to a `logging.Logger` to modules that opt in doesn't even seem unreasonable to me (so long as the root Logger is already handling streams and the like) and _something_ cats everything together when we turn out the lights.

Reply:
Take a look and tell me what you think. We are definitely using the `logging` module far more now.  Which was necessary to get the module-specific logging working.

Reply:
In the end, I moved as much logging logic out of our old logging class (`PrintLog`) as possible, and into a `logging.Logger` subclass.  I only had to keep something like the old `PrintLog` class in existence because it defines the API that people are using elsewhere in the code.